### DISCUSSION TITLE
Support for Reference Field Type in import via CSV/JSON

### DISCUSSION THREAD

**Discussion by I194** [AntlerVC/firetable#1187] [Support / Q&A] [answered]
Hello, am I right that there is no support for Reference field type in import via CSV/JSON? 

It's quite strange, because without it one should use API or set References manually. Moreover, this feature implemented in FlutterFlow data managment tool and it looks kinda easy to add (although I could be wrong, I am basing this assumption on [this answer](https://github.com/rowyio/rowy/discussions/1077#discussioncomment-4754056)).

**Comment by I194** [comment]
I found a solution. It's a bit inconvenient, but it works. Keep in mind that the solution described below is suitable only if you can generate the Reference in your table based on other columns in your table.

1. Import the table (.csv or .json)
2. During the import, for the field that should be of Reference type, select any other field that is a string (short/long text, for example)
3. Once the table is imported, change the field type to Derivative
4. Configure the Derivative (instructions can be found here: https://docs.rowy.io/field-types/derivative), and set the output type to Reference.
5. In the code editor, write something like this:
```
const derivative:Derivative = async ({row,ref,db,storage,auth,logging})=>{
  // WRITE YOUR CODE ONLY BELOW THIS LINE. DO NOT WRITE CODE/COMMENTS OUTSIDE THE FUNCTION BODY
  logging.log("derivative started")
  
  const id = row.yourExternalDocumentId;
  return db.doc('yourExternalDocument/' + id);
}
```
6. Click Update and then in the column settings, click the Evaluate all button.

**Reply by I194** [reply]
But the question of why the Reference type is not available when importing .csv and .json files still open.

**Comment by harinij** [comment, marked as answer]
@I194 Thanks for highlighting this, we have now added native import support for the Ref field and you can just import it directly from the CSV file without the need for the derivative. Hope this helps :)
